### PR TITLE
Global Styles: Use context when translating entries in theme.json

### PR DESCRIPTION
**Pull request by gziolo**
This PR attaches a translation context to the strings defined in `experimental-theme.json`, following the conversations at

- `block.json` – https://github.com/WordPress/wordpress-develop/pull/868#discussion_r556528330
- `theme.json` – https://github.com/WordPress/gutenberg/pull/27380#issuecomment-753879323

> 1. WP-CLI adds context to all the extracted strings (see https://github.com/wp-cli/i18n-command/blob/0fa24efa7460564d17fa4c6e1763e806eb1613c3/src/BlockExtractor.php#L43-L69 how it's done for `block.json` already
> 2. The code in WordPress that actually loads the `theme.json` (and also `block.json`) file will run all strings through `translate_with_gettext_context()` with the same context.

It uses an auxiliary structure defined https://github.com/WordPress/gutenberg/pull/28246#issuecomment-761117898 to know which data should be translated.

## How to test

- Verify that the PHP test passes.
- Install and activate the TT1-blocks theme and load the site editor. Verify that presets are working as expected and there're no errors in the logs.

### CODE REVIEW

File: lib/class-wp-theme-json-resolver.php [hunk 1]
```diff
@@ -140,8 +140,8 @@ private static function translate_presets( &$theme_json_structure, $domain = 'de
 						if ( empty( $item_to_translate[ $translatable_key ] ) ) {
 							continue;
 						}
-						// phpcs:ignore WordPress.WP.I18n.LowLevelTranslationFunction,WordPress.WP.I18n.NonSingularStringLiteralText,WordPress.WP.I18n.NonSingularStringLiteralDomain
-						$item_to_translate[ $translatable_key ] = translate( $item_to_translate[ $translatable_key ], $domain );
+						// phpcs:ignore WordPress.WP.I18n.LowLevelTranslationFunction,WordPress.WP.I18n.NonSingularStringLiteralText,WordPress.WP.I18n.NonSingularStringLiteralContext,WordPress.WP.I18n.NonSingularStringLiteralDomain
+						$item_to_translate[ $translatable_key ] = translate_with_gettext_context( $item_to_translate[ $translatable_key ], sprintf( 'theme %s %s', $path, $translatable_key ), $domain );
```
Comment (gziolo) [inline lib/class-wp-theme-json-resolver.php:144]: I haven't try to execute it yet. I'm more flagging this issue that wasn't addressed in the original PR.

@jorgefilipecosta, any recommendations on how to compute the context name based on the path?

**Comment by swissspidy** [comment]
Could you provide some examples of the expected contexts?

**Comment by jorgefilipecosta** [comment]
I guess we can add the contexts to file lib/experimental-i18n-theme.json.
Changing:
```
{
	"settings": {
		"typography": {
			"fontSizes": [
				"name"
			],
```

Into something like:
```
{
	"settings": {
		"typography": {
			"fontSizes": [ {
				"name": "Font size name",
			} ],
```

**Comment by nosolosw** [comment]
Hey, I wanted to share that I'm working on making `settings` and `styles` top-level keys of the theme.json in https://github.com/WordPress/gutenberg/pull/28110 as part of implementing https://github.com/WordPress/gutenberg/issues/28163 Not specifically related to this PR, but I wanted to surface it so you are aware as well. I presume I might need to reorganize the code (and potentially the `experimental-i18n-theme` shape), although I haven't looked at it yet.

**Comment by nosolosw** [comment]
:wave: I'm looking at this in the context of https://github.com/WordPress/gutenberg/issues/27506 so I'm going to push any commits directly if that's ok. First step: load my brain with the past conversations about this.

**Comment by nosolosw** [comment]
@swissspidy @aristath This is ready for review. Going to make https://github.com/WordPress/gutenberg/pull/28783 work based on these changes.

**Review by swissspidy — APPROVED** [review]

**Review by gziolo — COMMENTED** [review]
Nice work @nosolosw. 👍

**Pull request merged**